entry.go

```diff
@@ -78,8 +78,20 @@ func NewEntry(logger *Logger) *Entry {
 	}
 }
 
+func (entry *Entry) Dup() *Entry {
+	data := make(Fields, len(entry.Data))
+	for k, v := range entry.Data {
+		data[k] = v
+	}
+	return &Entry{Logger: entry.Logger, Data: data, Time: entry.Time, Context: entry.Context, err: entry.err}
```

Comment:
Why not copy `caller` into new entry? There are some scenarios where I want to pass the caller by myself, for example, many library accept a logger interface and use it to output internal log(such as SQL exec log for [`Gorm`](https://github.com/go-gorm/gorm/blob/26dd4c980a62d47c990a05da9e5566bff3b2b00c/logger/logger.go#L52)), so many users wrap logrus to the needed interface, but the caller got by `logrus` is inaccuracy, and I get the caller and pass it into `Entry`, and this change breaks my wrapped logger for these library.

Can I send a PR for copying caller in `Entry.Dup` util? What do you think of it? Thanks! @dgsb @epelc